domain/secret/service/exportimport.go

```diff
@@ -56,9 +57,14 @@ func (s *SecretService) GetSecretsForExport(ctx context.Context) (*SecretExport,
 			if rev.ValueRef != nil {
 				continue
 			}
-			data, _, err := s.secretState.GetSecretValue(ctx, md.URI, rev.Revision)
+			var data coresecrets.SecretData
+			err := s.secretState.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
```

Comment:
The entire operation to list and then get the secret values should be in a single txn.